core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/facts/ArtifactCoordinates.java

```diff
@@ -14,6 +14,7 @@
 public abstract class ArtifactCoordinates<T extends ArtifactCoordinates> implements ArtifactIdentifier<T> {
     public abstract String getName();
     public abstract String getVersion();
+    public abstract String getPurl();
```

Comment:
How do we plan to handle coordinates with insufficient information (e.g. with missing version or missing groupId)?

Reply:
How do we plan to handle coordinates types, which are not covered by the current purl definition? Throwing `UnsupportedOperationException`? Maybe add this to the interface?

Reply:
> How do we plan to handle coordinates types, which are not covered by the current purl definition?

Do you have any concrete example for such a type? Ideally we'd fix this by contributing to [purl-spec](https://github.com/package-url/purl-spec). But as an interim solution, yes, probably throw an exception.

Reply:
I think for example stackage is not supported. (Unimportant: I am even unsure how something like stackage would be supported, where the version is attached to stackage itself and references consistent package sets, instead of individual versions for every single component.)

Reply:
I do not think that missing information is an issue. In this case the component information is not sufficient, i.e., the purl is not valid and cannot be used. In this case, the artifact is not identified, so the compliance information has a major issue and therefore is only preliminary. The user has to solve that anyway.

The first usage of the purl is in the policy engine to make a unique hash of a policy violation, so that we can identify violations over different evaluations. I would prefer that the method always returns a String, even if the data is not complete. An unidentified artifact in the policy engine is a blocking violation, so the uniqueness of the violation issue is not that important, since it has to be solved.